bundler/lib/bundler/gem_helpers.rb

```diff
@@ -5,13 +5,11 @@ module GemHelpers
     GENERIC_CACHE = { Gem::Platform::RUBY => Gem::Platform::RUBY } # rubocop:disable Style/MutableConstant
     GENERICS = [
       [Gem::Platform.new("java"), Gem::Platform.new("java")],
-      [Gem::Platform.new("mswin32"), Gem::Platform.new("mswin32")],
-      [Gem::Platform.new("mswin64"), Gem::Platform.new("mswin64")],
-      [Gem::Platform.new("universal-mingw32"), Gem::Platform.new("universal-mingw32")],
-      [Gem::Platform.new("x64-mingw32"), Gem::Platform.new("x64-mingw32")],
-      [Gem::Platform.new("x86_64-mingw32"), Gem::Platform.new("x64-mingw32")],
-      [Gem::Platform.new("x64-mingw-ucrt"), Gem::Platform.new("x64-mingw-ucrt")],
-      [Gem::Platform.new("mingw32"), Gem::Platform.new("x86-mingw32")],
+      [Gem::Platform.new("windows"), Gem::Platform.new("windows")],
+      [Gem::Platform.new("mswin32"), Gem::Platform.new("windows")],
+      [Gem::Platform.new("mswin64"), Gem::Platform.new("windows")],
+      [Gem::Platform.new("mingw32"), Gem::Platform.new("windows")],
+      [Gem::Platform.new("mingw"), Gem::Platform.new("windows")],
```

Comment:
For the purposes of generics, I've introduced a new _virtual_ OS value `"windows"` which doesn't actually exist in the wild. Please let me know if I should map all these to "mingw" instead.